cluster_client.cpp

```diff
@@ -355,3 +408,41 @@ void cluster_client::create_request(struct timeval timestamp, unsigned int conn_
     }
 }
 
+void cluster_client::handle_response(unsigned int conn_id, struct timeval timestamp,
+                                     request *request, protocol_response *response) {
+    // handle "-MOVED"
+    if (response->is_error() &&
+        strncasecmp(response->get_status(), MOVED_MSG_PREFIX, MOVED_MSG_PREFIX_LEN) == 0) {
+        benchmark_debug_log("handle moved: %s %s\n", m_connections[conn_id]->get_address(), response->get_status());
+
+        // update stats
+        if (request->m_type == rt_get) {
+            m_stats.update_moved_get_op(&timestamp,
+                                        request->m_size + response->get_total_len(),
+                                        ts_diff(request->m_sent_time, timestamp));
+        } else if (request->m_type == rt_set) {
+            m_stats.update_moved_set_op(&timestamp,
+                                        request->m_size + response->get_total_len(),
+                                        ts_diff(request->m_sent_time, timestamp));
+        } else {
+            assert(0);
+        }
+
+        // connection already issued 'cluster slots' command, wait for slots mapping to be updated
+        if (m_connections[conn_id]->get_cluster_slots_state() != slots_done)
+            return;
+
+        // queue may stored uncorrected mapping indexes, empty them
+        key_index_pool empty_queue;
+        std::swap(*m_key_index_pools[conn_id], empty_queue);
+
+        // set connection to send 'CLUSTER SLOTS' command
```

Comment:
@YaacovHazan Is there a deliberate design decision to use `CLUSTER SLOTS` and not just update the specific slot mapping (as you already have enough information to do that)?